Objects/longobject.c

```diff
@@ -267,6 +282,21 @@ _PyLong_FromSTwoDigits(stwodigits x)
     return _PyLong_FromLarge(x);
 }
 
+static void
+int_dealloc(PyLongObject *op)
+{
+#if WITH_FREELISTS
```

Comment:
Should we move the `#if WITH_FREELISTS` into `PyInterpreterState_FreelistFree` to keep it localized?

Personally, I think we should just remove `WITH_FREELISTS`, but that needs a wider discussion.